GSoC 2023: Incremental build detection blog post (#6662)
jenkins-infra/jenkins.io: content/blog/2023/09/09/2023-09-09-incremental-build-detection-probe.adoc

---
layout: post
title: Incremental Build Detection Probe
tags:
- gsoc
- gsoc2023
- healthscore
- probes
- plugin
authors:
- jagruti
opengraph:
  image: /images/gsoc/2023/jagruti/incremental-build-detection-probe.png
sig:
- gsoc
---

== Introduction

> `Incremental Build Detection Probe`  was developed as a part of link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Google summer of Code 2023] for the plugin health scoring system project.

Reviewer: ```suggestion
The `Incremental Build Detection Probe` was developed as a part of link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Google Summer of Code 2023] for the plugin health scoring system project.
```
suggestion to adjust some syntax

> Incremental Builds aids in continuous delivery of the plugin. It deploys only the modules affected by new commits footnote:[https://www.cloudbees.com/jenkins/maven]. Jenkins core and plugin can be deployed incrementally rather than waiting for the developer to manually deploy the release footnote:[https://github.com/jenkinsci/jep/blob/master/jep/305/README.adoc].

Reviewer: ```suggestion
Incremental builds aid in continuous delivery of plugins, deploying only the modules affected by new commits footnote:[https://www.cloudbees.com/jenkins/maven].
Jenkins core and plugins can be deployed incrementally, rather than waiting for the developer to manually deploy the release footnote:[https://github.com/jenkinsci/jep/blob/master/jep/305/README.adoc].
```
suggestion for formatting and combining sentences

This probe identifies the plugins that have configured `incremental-builds`.

> == Importance of the Probe
> This probe encourages automation in delivery and saves maintainer from manual deployment.
> 

Reviewer: ```suggestion
== Importance of the Probe

This probe encourages delivery automation and saves the maintainer from manual deployment.

```
Is there any context as to why this is important? Saving effort/time is good, but expanding on why this probe is particularly important or helpful in that regard would be good to have.

> == Challenges
> Understanding the configuration and implementation posed a major challenge in this probe.
> To me, it seemed that I need to check the configuration for link:https://maven.apache.org/maven-ci-friendly.html[Maven Friendly CI versions].

Reviewer: ```suggestion
== Challenges

Understanding the configuration and implementation posed a major challenge in this probe.
To me, it seemed that I needed to check the configuration for link:https://maven.apache.org/maven-ci-friendly.html[Maven Friendly CI versions].
```
suggestion for tense update/syntax adjustments

My mentor pointed out the following and explained that link:https://matrix.to/#/!VkECGUHmVHbfLeicii:gitter.im/$_CwOuPxb9DPkkWTaO4QFIXOMmEqPeHpOq4Ca2eTrw-4?via=gitter.im&via=matrix.org&via=matrix.freyachat.eu[the actual configuration goes deeper]:

> - The link:https://github.com/jenkinsci/jep/blob/master/jep/305/README.adoc#setup[setup is actually done in `.mvn/maven.config`] it is essential that I look for the configuration in  `.mvn/maven.config`.

Reviewer: ```suggestion
- The link:https://github.com/jenkinsci/jep/blob/master/jep/305/README.adoc#setup[setup is actually done in `.mvn/maven.config`] and it is essential that I look for the configuration in `.mvn/maven.config`.
```
adjusting for syntax

> - Some plugins configure it in `extensions.xml` in the `.mvn` like the link:https://github.com/jenkinsci/kubernetes-plugin/blob/0141a5f145128b5b8458d40686a3904006f45b8d/.mvn/extensions.xml[kubernetes-plugin].

Reviewer: ```suggestion
- Some plugins configure it in `extensions.xml` in the `.mvn` section such as link:https://github.com/jenkinsci/kubernetes-plugin/blob/0141a5f145128b5b8458d40686a3904006f45b8d/.mvn/extensions.xml[kubernetes-plugin].
```
suggestion for syntax update

> - The final step was to look for the configuration in the CD workflow. As the `jep-229` (Continuous Delivery Probe) is already live, I need not look for this configuration.

Reviewer: ```suggestion
- The final step was to look for the configuration in the CD workflow.
As the `jep-229` (Continuous Delivery Probe) is already live, I did not need to look for this configuration.
```
suggestion for formatting/syntax

== Outcome and Learning

> Learned more about GitHub Workflows and its strength. This probe is merged and ready to be live.

Reviewer: ```suggestion
I learned more about GitHub Workflows and its strength.
This probe is merged and ready to be live.
```
suggestion for syntax/formatting

== Links

> For more information or if you have any questions, please visit the official GSoC 2023 project link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to  "Plugin Health Score"] description page.

Reviewer: ```suggestion
For more information or if you have any questions, please visit the official GSoC 2023 project link:/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Adding Probes to "Plugin Health Score"] description page.
```
removing unneeded space
